File: typescript/namespaceTest.ts

namespace nsa{
	export function fa():string{
		return "apple";
	}
}

namespace nsb{
	export function fa():string{
		return "banana";
	}
}

function test(){
	console.log(nsa.fa())
	console.log(nsb.fa())
}

test()

File: c#/namespaceTest.cs
using System;

namespace A{
	public class Sayer{
		public void say(){
			Console.WriteLine("apple");
		}
	}
}

namespace B{
	public class Sayer{
		public void say(){
			Console.WriteLine("banana");
		}
	}
}

namespace TestApplication{
	class Test{
		static void Main(string[] args){
			new A.Sayer().say();
			new B.Sayer().say();
		}
	}
}
